wptherml/exciton.py

```diff
@@ -0,0 +1,594 @@
+import numpy as np
+from matplotlib import pyplot as plt
+from .spectrum_driver import SpectrumDriver
+
+
+class ExcitonDriver(SpectrumDriver):
+    """A class for computing the dynamics and spectra of a system modelled by the Frenkel Exciton Hamiltonian
+
+    Attributes
+    ----------
+    radius : float
+        the radius of the sphere
+
+
+    Returns
+    -------
+    None
+
+    Examples
+    --------
+    >>> fill_in_with_actual_example!
+    """
+
+    def __init__(self, args):
+        self.parse_input(args)
+        # print("Exciton Energy is  ", self.exciton_energy)
+        # allocate the exciton Hamiltonian
+
+        self.exciton_hamiltonian = np.zeros(
+            (self.number_of_monomers, self.number_of_monomers)
+        )
+        self.c_vector = np.zeros(
+            (self.number_of_monomers, 1),  dtype=complex
+        )  # <== wavefunction coefficient vector
+        
+        # define cartesian coordinates of each monomer
+        self._compute_cartesian_coordinates()
+
+        # build exciton Hamiltonian
+        self.build_exciton_hamiltonian()
+
+        # Probably want to allow the user to specify an initial state!
+        # but right now just have the initial state with exciton localized on site 1
+        self.c_vector[0,0] = 1 + 0j
+        #self.density_matrix = np.dot(self.c_vector, np.conj(self.c_vector.T))
+        self.density_matrix = np.dot(self.c_vector, np.conj(self.c_vector.T))
+
+    def parse_input(self, args):
+        if "exciton_energy" in args:
+            self.exciton_energy = args["exciton_energy"]
+        else:
+            self.exciton_energy = 0.5
+
+        #############################################################
+        # going to change this to aggregate_shape 
+        #if "number_of_monomers" in args:
+        #    self.number_of_monomers = args["number_of_monomers"]
+        #else:
+        #    self.number_of_monomers = 2
+        #############################################################
+
+        if "aggregate_shape" in args:
+            self.aggregate_shape = args["aggregate_shape"]
```

Comment:
@lanetolley I added this input flag which is meant to be a tuple specifying the shape of the aggregate in terms of number of repeats of the monomer along the x, y, and z axis. For example, a 2D aggregate with 4 total monomers, 2 monomers along the x-axis and 2 along the y-axis, would be specified like (2, 2, 1).  A 3D aggregate with 8 total monomers, 2 along x, 2 along y, 2 along z, would be (2, 2, 2), etc.